users/users_test.go

```diff
@@ -51,6 +51,64 @@ func TestValidate(t *testing.T) {
 			},
 			err: users.ErrMalformedEntity,
 		},
+		"validate user with utf8 email (cyrillic)": {
+			user: users.User{
+				Email:    "почта@кино-россия.рф",
+				Password: password,
+			},
+			err: nil,
+		},
+		"validate user with utf8 email (hieroglyph)": {
+			user: users.User{
+				Email:    "艾付忧西开@艾付忧西开.再得",
+				Password: password,
+			},
+			err: nil,
+		},
+		"validate user with no email tld": {
+			user: users.User{
+				Email:    "user@example.",
+				Password: password,
+			},
+			err: users.ErrMalformedEntity,
+		},
+		"validate user with too long email tld": {
+			user: users.User{
+				Email:    "user@example.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
```

Comment:
This and specially the next one are too long. You need a function here. We already did in other places things like:
```
letters = "abcdefghijklmnopqrstuvwxyz"
func randomString(n int) string {
	b := make([]byte, n)
	for i := range b {
		b[i] = letters[rand.Intn(len(letters))]
	}
	return string(b)
}

longDomain := randomString(maxDomainLen + 1)
```

Something like that